mono/metadata/object.c

```diff
@@ -7195,21 +7195,17 @@ mono_string_to_utf16 (MonoString *s)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
-	char *as;
-
 	if (s == NULL)
 		return NULL;
 
-	as = (char *)g_malloc ((s->length * 2) + 2);
-	as [(s->length * 2)] = '\0';
-	as [(s->length * 2) + 1] = '\0';
-
-	if (!s->length) {
-		return (gunichar2 *)(as);
+	int const length = s->length;
+	gunichar2 * const as = (gunichar2 *)g_malloc ((length + 1) * sizeof (gunichar2));
```

Comment:
Why use a `gunichar2` if you are returning a `mono_unichar2`?

Reply:
The type was already in use within the function -- at least on the return, and it is shorter, so I favored it for precedent and terseness. But ok I removed all gunichar2 from the function. And replaced some sizeof(type) with sizeof(*pointer_variable).